### PR TITLE
Added basic CRUD operation index operation for configurations

### PR DESCRIPTION
**Pull request by akbhatta**
### Description
Added basic CRUD operation index operation for configurations

[Tests]
Added Integration tests for CRUD operations

Other changes:
JSON tags are changed from CamelCase to snake_case to be consistent with other plugins

Signed-off-by: @akbhatta 

### Issues Resolved
NA

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+        }
+        return response.result == DocWriteResponse.Result.UPDATED
+    }
+
+    /**
+     * delete NotificationConfigDoc for given id
+     * @param id the id for the document
+     * @return true if successful, false otherwise
+     */
+    fun deleteNotificationConfig(id: String): Boolean {
```
Comment (zhongnansu) [inline notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt:259]: Np: do we also need a bulk delete API and function?

File: notifications/src/main/kotlin/org/opensearch/notifications/model/SendMessageRequest.kt [hunk 2]
```diff
@@ -59,8 +59,8 @@ internal class SendMessageRequest : ActionRequest, ToXContentObject {
         private const val REF_TAG_FIELD = "refTag"
```
Comment (zhongnansu) [inline notifications/src/main/kotlin/org/opensearch/notifications/model/SendMessageRequest.kt:59]: also change this to snakecase? But I think this refTag field will be removed, right? It belongs to the old implemetation.

**Review by zhongnansu — APPROVED** [review]
LGTM, left some comments

File: notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+                doc.metadata.createdTime,
+                doc.metadata.tenant,
+                doc.config
+            )
+        }
+    }
+
+    /**
+     * Initialize the class
+     * @param client The ES client
```
[the remark this answers is not in the document]
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt:93]: fixed

File: notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt (hunk 1 above)
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt:259]: will add it in next commit if required

File: notifications/src/main/kotlin/org/opensearch/notifications/model/SendMessageRequest.kt (hunk 2 above)
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/notifications/model/SendMessageRequest.kt:59]: fixed

**Review by anirudha — APPROVED** [review]
lg

**Pull request merged**